### ISSUE TITLE
IEEE 802.1x authenticated host is unable to share the internet

### ISSUE THREAD

**Issue by RVFET** [Genymobile/gnirehtet#550]
When the host (pc) is connected to a 802.1x authenticated network, shared connection does not work. Connection between PC and the android device establishes successfully but no internet. I assume it's because the restrictions of 802.1x, but is there a way to overcome that?

**Comment by VaradGupta23** [comment]
You're absolutely right: IEEE 802.1X networks (typically enterprise or university networks) use port-based access control with strict authentication tied to a single MAC address or device. That’s why:

Your PC is allowed access (authenticated via 802.1X),

But when you try to share internet (via Gnirehtet, ICS, or other methods) to your Android phone, the traffic either gets blocked or doesn't route at all.

 Why Sharing Fails on 802.1X Networks
802.1X typically binds access to your PC’s MAC address — not your Android device or any virtual interface (e.g., from gnirehtet).

If the network sees packets from a new MAC or another IP (like your Android's VPN IP 10.0.0.2), it drops them or redirects them to a captive portal.

Workarounds to Bypass 802.1X Restrictions
Option 1: Use Gnirehtet with NAT (Best for Android)
Gnirehtet already uses a VPN-based reverse tethering method where:

All Android traffic is routed through the PC (not directly from the device),

The host appears to be the only device on the network.

If internet still doesn’t work in this setup, your 802.1X network may block specific outbound traffic.

Try:

gnirehtet run
If not working, go to Option 2.

 Option 2: Use a SOCKS Proxy or VPN on PC
Tunnel all outgoing traffic from Android through your PC using tools like:

redsocks or tsocks + iptables (Linux)

Proxifier (Windows)

Or setup a local VPN server on the PC

This ensures all device traffic looks like it's coming from the PC, not from the Android itself.

 Option 3: MAC Spoofing (Advanced / Risky)
Some setups allow internet sharing only if:

Both the PC and shared device use the same MAC.

You can try spoofing the PC’s MAC address on your Android device — or vice versa — but:

 Warning: Many 802.1X networks actively detect and block spoofed MACs and may disconnect you entirely or even flag you.

Option 4: Bridge Mode (Not Recommended with 802.1X)
Normally, ICS (Internet Connection Sharing) via bridge mode fails because 802.1X doesn’t allow multiple MACs. Even if you bridge interfaces, the Android device’s MAC becomes visible → traffic blocked.

 Option 5: Run a Full VPN (e.g., WireGuard)
Set up WireGuard VPN server on your PC (or somewhere trusted),

Connect Android to PC over gnirehtet,

On Android, run WireGuard VPN client with PC as the exit node.

This way:

PC handles VPN tunneling,

Android only needs a local connection,

No unauthorized traffic hits 802.1X network directly.

**Comment by RVFET** [comment]
@VaradGupta23  The `—`, `’`, `→` characters and overall structure screams AI-generated content. While I appreciate the response, I could have asked an AI myself if I wanted that. I'm looking for input from actual developers/community members, not AI guidance.
